services/nodeping/nodeping-status.service.js

```diff
@@ -0,0 +1,114 @@
+'use strict'
+
+const { BaseJsonService } = require('..')
+const Joi = require('joi')
+
+const rowSchema = Joi.object().keys({ su: Joi.boolean() })
+
+const schema = Joi.array()
+  .items(rowSchema)
+  .min(1)
+
+/*
+ * this is the checkUuid for the NodePing.com (as used on the [example page](https://nodeping.com/reporting.html#results))
+ */
+const sampleCheckUuid = 'jkiwn052-ntpp-4lbb-8d45-ihew6d9ucoei'
+
+class NodePingStatus extends BaseJsonService {
+  static get category() {
+    return 'monitoring'
+  }
+
+  static get defaultBadgeData() {
+    return {
+      label: 'Status',
+    }
+  }
+
+  static get route() {
+    return {
+      base: 'nodeping/status',
+      pattern: ':checkUuid',
+      queryParams: [
+        'label',
+        'up_message',
+        'down_message',
+        'up_color',
+        'down_color',
+      ],
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'NodePing status',
+        namedParams: {
+          checkUuid: sampleCheckUuid,
+        },
+        staticPreview: this.render({ status: true }),
+      },
+      {
+        title: 'NodePing status (customized)',
+        namedParams: {
+          checkUuid: sampleCheckUuid,
+        },
+        queryParams: {
+          label: 'Status',
+          up_message: 'Online',
+          up_color: 'lightgrey',
+          down_message: 'offline',
+          down_color: 'orange',
+        },
+        staticPreview: this.render({
+          status: true,
+          label: 'Status',
+          upMessage: 'Online',
+          upColor: 'lightgrey',
+        }),
+      },
+    ]
+  }
+
+  async fetch({ checkUuid }) {
+    const rows = await this._requestJson({
+      schema,
+      url: `https://nodeping.com/reports/results/${checkUuid}/1`,
+      options: {
+        method: 'GET',
+        qs: { format: 'json' },
+        headers: {
+          'cache-control': 'no-cache',
+        },
+      },
+    })
+    return { status: rows[0].su }
+  }
+
+  async handle(
+    { checkUuid },
+    {
+      up_message: upMessage,
+      down_message: downMessage,
+      up_color: upColor,
+      down_color: downColor,
+    }
+  ) {
+    const { status } = await this.fetch({ checkUuid })
+    return this.constructor.render({
+      status,
+      upMessage,
+      downMessage,
+      upColor,
+      downColor,
+    })
+  }
+
+  static render({ status, label, upMessage, downMessage, upColor, downColor }) {
+    return status
+      ? { label, message: upMessage || 'up', color: upColor || 'brightgreen' }
```

Comment:
We should also have some tests that validate this logic. If you add the query param (`?style=_shields_test`) to the test `get` call then you can check the `color` on the json response

Reply:
I tried to run coverage tests, but the command in the tutorial ("npm run coverage:test:services -- --only=nodeping") does not like the only=nodeping ('npm WARN invalid config only="nodeping"'), so it tries to run all the tests, and some (before nodeping) fail.

Reply:
Yeah it needs a second `--` now, we'll need to update the tutorial accordingly. 

Try `npm run coverage:test:services -- -- --only=nodeping && npm run coverage:report`

Reply:
That worked.  But I don't see where the uncovered branches are...

Reply:
You might avoid the problem by using default parameters:

```js
static render({ status, upMessage = 'up', downMessage = 'down', …
```

Reply:
> That worked. But I don't see where the uncovered branches are...

What we should strive for are tests that validate all expected behaviors, which was the intent of my comment. It's not just about code coverage.

We've got default values being defined for the message and the color, with user definable values that will override those defaults. We'd want tests that validate those default values are applied as expected, and that the user-defined values are indeed used